src/java.base/share/classes/jdk/internal/classfile/impl/BufferedCodeBuilder.java

```diff
@@ -60,11 +57,10 @@ public BufferedCodeBuilder(MethodInfo methodInfo,
         this.context = context;
         this.startLabel = new LabelImpl(this, -1);
         this.endLabel = new LabelImpl(this, -1);
-        this.original = original;
         this.methodInfo = methodInfo;
         this.maxLocals = Util.maxLocals(methodInfo.methodFlags(), methodInfo.methodTypeSymbol());
-        if (original != null)
-            this.maxLocals = Math.max(this.maxLocals, original.maxLocals());
```

Comment:
`original::maxLocals`set the counter for `CodeBuilder::allocateLocal`
By restricting calculation of maxLocals to "origin instanceof CodeAttribute" may cause invalid locals allocations for chained code builders. The number might not be exposed in the API, however we need to know it internally.

Reply:
I think we should just do `Util.maxLocals` without this `origin` check. Your said problem can happen too if a DirectMethodBuilder transforms a BufferedCodeBuilder.Model.

Note that any code builder can receive a code model half way with `cob.transform(anyModel, CodeTransform.ACCEPT_ALL)`, which the buffered code builder initialization also uses.
I think we should just ask users to use `CodeLocalsShifter` in that case, especially if the transform is inserting new locals. We cannot buffer existing chained builders for performance reasons.

Reply:
Updated this to internal API that accesses the existing local variable management on BufferedCodeBuilder. We can discuss this consistency problem later.